pkg/vmprovider/providers/vsphere/virtualmachine/backup.go

```diff
@@ -35,51 +35,102 @@ func BackupVirtualMachine(
 	vmCtx context.VirtualMachineContext,
 	vcVM *object.VirtualMachine,
 	bootstrapData map[string]string) error {
-	vmKubeData, err := getEncodedVMKubeData(vmCtx.VM)
+
+	resVM := res.NewVMFromObject(vcVM)
```

Comment:
I'd eventually like to get rid of this resources VM wrapper - prefer to use the govmomi `vcVM.Properties()` directly here.

Reply:
Updated as suggested in the latest commit.